backend/spec/controllers/spree/admin/orders/customer_details_controller_spec.rb

```diff
@@ -20,23 +20,175 @@
+            it { expect(controller).to receive(:guest_checkout?).twice.and_call_original }
+            it { expect(controller).to receive(:load_user).and_call_original }
+            after { send_request(_attributes) }
+          end
+        end
+
+        context 'having invalid parameters' do
+          before do
+            allow(Spree.user_class).to receive(:find_by).and_return(false)
+            allow(controller).to receive(:guest_checkout?).and_return(false)
+          end
+
+          context 'having failure response' do
+            before { send_request(_attributes) }
+            it { expect(response).to render_template(:edit) }
+          end
+
+          context 'with correct method flow' do
+            it { expect(order).to_not receive(:update_attributes).with(_attributes[:order]) }
+            it { expect(controller).to receive(:load_order).and_call_original }
+            it { expect(controller).to receive(:guest_checkout?).and_call_original }
+            it { expect(controller).to receive(:load_user).and_call_original }
+            after { send_request(_attributes) }
+          end
+        end
+
+        describe '#load_user' do
+          context 'having valid parameters' do
+            before do
+              allow(Spree.user_class).to receive(:find_by).and_return(user)
+              allow(order).to receive_messages(update_attributes: true)
+              allow(order).to receive_messages(next: false)
+              allow(order).to receive_messages(complete?: true)
+              allow(order).to receive_messages(refresh_shipment_rates: true)
+              allow(order).to receive_messages(associate_user!: true)
+              allow(controller).to receive(:guest_checkout?).and_return(false)
+              allow(order).to receive(:associate_user!)
+            end
+
+            it "expects to assign user" do
+              send_request(_attributes)
+              expect(assigns[:user]).to eq(user)
+            end
+
+            context 'with correct method flow' do
+              it { expect(Spree.user_class).to receive(:find_by).with(id: user.id.to_s).and_return(user) }
+              after { send_request(_attributes) }
+            end
+          end
+
+          context 'with invalid parameters' do
+            before do
+              allow(Spree.user_class).to receive(:find_by).and_return(nil)
+              allow(controller).to receive(:guest_checkout?).and_return(false)
+            end
+
+            it "expects to not assign user" do
+              send_request(_attributes)
+              expect(assigns[:user]).to_not eq(user)
+            end
+
+            context 'with correct method flow' do
+              it { expect(Spree.user_class).to receive(:find_by).with(id: user.id.to_s).and_return(nil) }
+              it { expect(Spree.user_class).to receive(:find_by).with(email: _attributes[:order][:email])
+                .and_return(nil) }
```

Comment:
Expression at 187, 34 should be on its own line.<br>Place the . on the previous line, together with the method call receiver.